.github/workflows/lint.yml

```diff
@@ -0,0 +1,107 @@
+
+jobs:
+  # Determine what the js changes are
+  js-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      packages: ${{ steps.filter.outputs.changes }}
+    steps:
+      # Use paths-filter to determine which paths have changes
+      # https://github.com/dorny/paths-filter
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          # Format of key: value. When there is a change detected in value, the
+          # key will be returned in the output.
+          filters: |
+            packages/dashboard: packages/dashboard/**
+            packages/dashboard-v2: packages/dashboard-v2/**
+            packages/dnslink-api: packages/dnslink-api/**
+            packages/handshake-api: packages/handshake-api/**
+            packages/health-check: packages/health-check/**
+            packages/website: packages/website/**
+
+  # Determine what the dockerfile changes are
+  dockerfile-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      packages: ${{ steps.filter.outputs.changes }}
+    steps:
+      # Use paths-filter to determine which paths have changes
+      # https://github.com/dorny/paths-filter
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          # Format of key: value. When there is a change detected in value, the
+          # key will be returned in the output.
+          filters: |
+            docker/nginx/Dockerfile: docker/nginx/Dockerfile
+            docker/sia/Dockerfile: docker/sia/Dockerfile
+            packages/dashboard/Dockerfile: packages/dashboard/Dockerfile
+            packages/dnslink-api/Dockerfile: packages/dnslink-api/Dockerfile
+            packages/handshake-api/Dockerfile: packages/handshake-api/Dockerfile
+            packages/health-check/Dockerfile: packages/health-check/Dockerfile
+            packages/website/Dockerfile: packages/website/Dockerfile
+
+  # Run the js lint if there are changes to any of the js packages
+  js-lint:
+    needs: js-changes
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        working-directory: ${{ fromJSON(needs.js-changes.outputs.packages) }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run JS-Lint Action
+        uses: ./.github/actions/js-lint
+        with:
+          working-directory: ${{ matrix.working-directory }}
+
+  # Run the hadolint dockefile lint if there are changes to any of the
+  # dockerfiles
+  hadolint:
+    needs: dockerfile-changes
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        dockerfile: ${{ fromJSON(needs.dockerfile-changes.outputs.packages) }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hadolint/hadolint-action@v2.0.0
+        with:
+          dockerfile: ${{ matrix.dockerfile }}
+
+  # Determine if there are python changes
+  python-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      # Check if the changed status of py is true
+      updates: steps.changes.outputs.py == 'true'
```

Comment:
I'd like to improve some consistency, sometimes the output is called "packages" sometimes "updates". The id of the step is sometimes called "changes" and sometimes "filter". While those are all valid, I think it makes it easier to read the file if we keep consistent naming, I actually had to read up on some docs to find out whether those are different for a reason.